### ISSUE TITLE
Validation pipeline fails on PDB-Dev 8 and 66

### ISSUE THREAD

**Issue by aozalevsky** [salilab/IHMValidation#29]
Parsing of the PDB-Dev entries 8 and 66 fails with the following trace:

```
Traceback (most recent call last):
  File "/IHMValidation/example/Execute.py", line 203, in <module>
    report = WriteReport(args.f)
  File "/IHMValidation/example/../master/pyext/src/validation/Report.py", line 26, in __init__
    self.input = GetInputInformation(self.mmcif_file)
  File "/IHMValidation/example/../master/pyext/src/validation/__init__.py", line 32, in __init__
    self.system, = ihm.reader.read(fh, model_class=self.model)
  File "/root/miniforge/lib/python3.9/site-packages/ihm/reader.py", line 3260, in read
    more_data = r.read_file()
  File "/root/miniforge/lib/python3.9/site-packages/ihm/format.py", line 589, in read_file
    return self._read_file_c()
  File "/root/miniforge/lib/python3.9/site-packages/ihm/format.py", line 640, in _read_file_c
    eof, more_data = _format.ihm_read_file(self._c_format)
  File "/root/miniforge/lib/python3.9/site-packages/ihm/reader.py", line 1247, in __call__
    a.append(self.sysr.ranges.get(obj, entity_poly_segment_id))
  File "/root/miniforge/lib/python3.9/site-packages/ihm/reader.py", line 191, in get
    return asym_or_entity(*self._id_map[range_id])
  File "/root/miniforge/lib/python3.9/site-packages/ihm/__init__.py", line 1289, in __call__
    return AsymUnitRange(self, seq_id_begin, seq_id_end)
  File "/root/miniforge/lib/python3.9/site-packages/ihm/__init__.py", line 1198, in __init__
    raise TypeError("Can only create ranges for polymeric entities")
TypeError: Can only create ranges for polymeric entities
```

The parsing code seems to be quite generic:
https://github.com/salilab/IHMValidation/blob/9d767d1ad6fc669661ee39a56db9212ae5be751d/master/pyext/src/validation/__init__.py#L29-L35

So I presume the problem is indeed in the cif files. PDB-Dev 8 has this in the header:

```
#
loop_
_entity.id
_entity.type
_entity.src_method
_entity.pdbx_description
_entity.formula_weight
_entity.pdbx_number_of_molecules
_entity.details
1 polymer man chr2L_60-161 ? 1 ?
#
<...>
loop_
_struct_asym.id
_struct_asym.entity_id
_struct_asym.details
A 1 chr2L_60-161
#
```

Which I guess causes failure at the `asym.entity` check from the `python-ihm` https://github.com/ihmwg/python-ihm/blob/0989b68412c01359e9f51aaf8413325532306737/ihm/__init__.py#L1196-L1201

@benmwebb I guess I need your advice on this: is this an actual artifact in the cif or it should be handled in the code?

**Comment by benmwebb** [comment]
Ah, I bet this is caused by ihmwg/python-ihm#70 - easy to check for by downgrading python-ihm to 0.27 to see if the problem goes away. PDB-Dev 8 has always been problematic because it does not provide a primary sequence. The proper fix in python-ihm would be to allow the user to override whether an entity is a polymer or not (usually it guesses based on the length of the sequence). In this case you know it's a polymer even though length==1 because the `entity` table in the input file says so.

Same deal with PDB-Dev 66 where entity 4 is a single GLY residue. This would normally be considered a nonpolymer but they have marked it as a polymer. This seems like a mistake to me but we should handle it better in python-ihm.

**Comment by benmwebb** [comment]
@aozalevsky, did this commit to python-ihm fix this issue for you? @brindakv will look at PDB-Dev 66 to try to properly fix entity 4, but python-ihm *should* work with both 8 and 66 now. If it does, I'll make a new python-ihm release shortly so that you can just pull the latest version into your docker/podman/singularity container.

**Comment by aozalevsky** [comment]
yep, now both structures can be parsed, and reports seem fine. I guess the issue could be closed. Or better wait for python-ihm release to link in the issue?

**Comment by aozalevsky** [comment]
Fix released in [python-ihm-0.30](https://github.com/ihmwg/python-ihm/releases/tag/0.30)
